mysql.go

```diff
@@ -306,6 +306,28 @@ func (dialector Dialector) DataTypeOf(field *schema.Field) string {
 		return dialector.getSchemaBytesType(field)
 	}
 
+	if field.AutoIncrement {
```

Comment:
Change this to 

```go
if field.AutoIncrement {
  return dialector.getSchemaIntAndUnitType(field)
}
```
?

Reply:
if specified the type already, just returns `field.DataType + " AUTO_INCREMENT"` sounds better

Reply:
ParseField handles custom types

if val, ok := field.TagSettings["TYPE"]; ok {
		switch DataType(strings.ToLower(val)) {
		case Bool, Int, Uint, Float, String, Time, Bytes:
			field.DataType = DataType(strings.ToLower(val))
		default:
			field.DataType = DataType(val)
		}
	}

Reply:
DataTypeOf processing field types cannot match custom types
switch field.DataType {
	case schema.Int, schema.Uint:
		return dialector.getSchemaIntAndUnitType(field)

Reply:
if field.AutoIncrement {
    return  string(field.DataType) + " AUTO_INCREMENT"
}
This right?

Reply:
```go
if !strings.Contains(strings.ToLower(string(field.DataType)), " auto_increment")
```

Reply:
We only handle with 7 types, if not, it means that the user has defined type in the [tag](https://github.com/go-gorm/gorm/blob/master/schema/field.go#L306), just add `default` into switch to handle it, which may be called `dialector.getSchemaCustomType(field)`, and we can improve it later if necessary.

Reply:
autoIncrement := field.AutoIncrement
	if !autoIncrement {
		_, autoIncrement = field.TagSettings["AUTO_INCREMENT"]
	}
	if autoIncrement {
		return string(field.DataType) + " AUTO_INCREMENT"
	}
A better way is to use string(field.datatype)

Reply:
https://github.com/go-gorm/gorm/pull/5452
https://github.com/go-gorm/gorm/issues/5299
now you need to implement AUTO_INCREMENT

Reply:
@ruanlang  jinzhu means if it is not the custom types, it will be marked `AUTO_INCREMENT` tag by `dialector.GetSchemaIntAndUnitType `, so you need to check whether there have been a `AUTO_INCREMENT`, I means you can move it to 
```
switch {
...
default:
// implement
```

Reply:
@a631807682 fixed